src/app/config/dapps/index.ts

```diff
@@ -0,0 +1,36 @@
+import { Layer } from '../../../oasis-nexus/api'
+import { Network } from '../../../types/network'
+import { wRose } from './wrose'
+import { TFunction } from 'i18next'
+
+export type DappReference = {
+  description: string
+  label: string
+  url: string
+}
+
+export type DappReferenceGenerator = (t: TFunction) => DappReference
+
+const dappsForToken: Record<Network, Partial<Record<Layer, Record<string, DappReferenceGenerator>>>> = {
+  [Network.testnet]: {
+    [Layer.sapphire]: {
+      [wRose.testnetAddress]: wRose.app,
+    },
+  },
+  [Network.mainnet]: {
+    [Layer.sapphire]: {
+      [wRose.mainnetAddress]: wRose.app,
+    },
+  },
+}
+
+/**
+ * Return a dApp relevant to a given address, what we want to advertise
+ */
+export const getDappForAddress = (t: TFunction, network: Network, layer: Layer, address: string) => {
+  const networkData = dappsForToken[network]
+  const layerData = networkData ? networkData[layer] : undefined
+  const generator = layerData ? layerData[address] : undefined
```

Comment:
I would go with optional chaining here instead of variables and ternary operators  `const generator = dappsForToken?.[network]?.[layer]?.[address]`

Reply:
Oh cool, I didn't know about this version of syntax. Fixed.